spec/bolt/shell/bash_spec.rb

```diff
@@ -0,0 +1,190 @@
+        expect(result).to be_empty
+        expect(stdin.string).to eq("my-password\n")
+      end
+
+      it "uses password as the default sudo-password" do
+        inventory.set_config(target, target.transport, 'password' => 'my-password')
+        result = shell.handle_sudo(stdin, shell.sudo_prompt, "")
+        expect(result).to be_empty
+        expect(stdin.string).to eq("my-password\n")
+      end
+
+      it "fails if no sudo password is set" do
+        err = /Sudo password.*was not provided/
+        expect { shell.handle_sudo(stdin, shell.sudo_prompt, "") }.to raise_error(Bolt::Node::EscalateError, err)
+      end
+    end
+
+    it "writes stdin if the sudo id is encountered" do
+      uuid = SecureRandom.uuid
+      allow(SecureRandom).to receive(:uuid).and_return(uuid)
+
+      result = shell.handle_sudo(stdin, uuid, "hello world")
+      expect(result).to be_empty
+      expect(stdin.string).to eq("hello world\n")
+    end
+
+    it "raises an error if the user is not in sudoers" do
+      msg = "root is not in the sudoers file."
+      err = /User root does not have sudo permission/
+      expect { shell.handle_sudo(stdin, msg, "") }.to raise_error(Bolt::Node::EscalateError, err)
+    end
+
+    it "raises an error if the password is wrong" do
+      msg = "Sorry, try again."
+      err = /Sudo password for user root not recognized/
+      expect { shell.handle_sudo(stdin, msg, "") }.to raise_error(Bolt::Node::EscalateError, err)
+    end
+
+    it "returns the input string if there's no error" do
+      line = "This is just ordinary output"
+      expect(shell.handle_sudo(stdin, line, "")).to eq(line)
+    end
+  end
+
+  describe "#run_command" do
+    it "runs a command" do
+      expect(connection).to receive(:execute).with('echo hello world')
+      shell.run_command('echo hello world')
+    end
+
+    it "runs a command as the run-as user set on the target" do
+      inventory.set_config(target, target.transport, 'run-as' => 'soandso')
+      expect(connection).to receive(:execute).with(/sudo .* -u soandso .* whoami/)
+      shell.run_command('whoami')
+    end
+
+    it "runs a command as the run-as user passed as an option" do
+      inventory.set_config(target, 'ssh', 'run-as' => 'soandso')
+      expect(connection).to receive(:execute).with(/sudo .* -u suchandsuch .* whoami/)
+
+      shell.run_command('whoami', run_as: 'suchandsuch')
+    end
+  end
+
+  describe "#execute" do
+    it "returns stdout, stderr and exitcode" do
+      execute_result = mock_result(stdout: "hello world", stderr: "some errors", exitcode: 6)
+      expect(connection).to receive(:execute).and_return(execute_result)
+
+      result = shell.execute('my cool command')
+      expect(result.stdout.string).to eq("hello world")
+      expect(result.stderr.string).to eq("some errors")
+      expect(result.exit_code).to eq(6)
+    end
+
+    it "runs a command with extremely long UTF-8 output" do
+      stdout = "hello ☃" * 60000
```

Comment:
I added a similar test to the local integration tests, and it looks like this case doesn't actually break without this change? 
```
# spec/integration/transport/local_spec.rb
  it "long input" do
    stdin = "hello ☃" * 60000
    expect(runner.run_command(target, "echo #{stdin}").value['stdout'].strip).to eq(stdin)
  end 
```

I'm not sure how to reproduce the error that this change fixes.

Reply:
Your example isn't passing a large value over stdin, it's running a large command line string.